finality-aleph/src/aggregation/mod.rs

```diff
@@ -1,14 +1,40 @@
-use aleph_bft::SignatureSet;
-use aleph_bft_rmc::Message;
+use std::{fmt::Debug, hash::Hash, marker::PhantomData, time::Instant};
+
+use aleph_aggregator::NetworkError;
+use aleph_bft::{Recipient, SignatureSet};
 use sp_runtime::traits::Block;
 
-use crate::crypto::Signature;
+use crate::{
+    crypto::Signature,
+    metrics::Checkpoint,
+    network::{Data, DataNetwork, SendError},
+    Metrics,
+};
 
-mod aggregator;
-mod multicast;
+pub type RmcNetworkData<B> =
+    aleph_aggregator::RmcNetworkData<<B as Block>::Hash, Signature, SignatureSet<Signature>>;
 
-pub use aggregator::{BlockSignatureAggregator, IO};
-pub use multicast::SignableHash;
+pub struct NetworkWrapper<D: Data, N: DataNetwork<D>>(pub N, pub PhantomData<D>);
 
-pub type RmcNetworkData<B> =
-    Message<SignableHash<<B as Block>::Hash>, Signature, SignatureSet<Signature>>;
+impl<H: Debug + Hash + Eq + Debug + Copy> aleph_aggregator::Metrics<H> for Metrics<H> {
+    fn report_aggregation_complete(&mut self, h: H) {
+        self.report_block(h, Instant::now(), Checkpoint::Aggregating);
+    }
+}
+
+#[async_trait::async_trait]
+impl<T, D> aleph_aggregator::DataNetwork<D> for NetworkWrapper<D, T>
```

Comment:
Yeah, this implementation is kinda a farce. :/ See comment by the trait definition.